src/app/CommandHandlerImpl.cpp

```diff
@@ -135,7 +135,16 @@ CHIP_ERROR CommandHandlerImpl::TryAddResponseData(const ConcreteCommandPath & aR
 
     TLV::TLVWriter * writer = GetCommandDataIBTLVWriter();
     VerifyOrReturnError(writer != nullptr, CHIP_ERROR_INCORRECT_STATE);
-    ReturnErrorOnFailure(aEncodable.EncodeTo(*writer, TLV::ContextTag(CommandDataIB::Tag::kFields)));
+    auto context = GetExchangeContext();
+    if (context && context->HasSessionHandle())
+    {
+        ReturnErrorOnFailure(aEncodable.EncodeTo(*writer, TLV::ContextTag(CommandDataIB::Tag::kFields),
+                                                 context->GetSessionHandle()->GetFabricIndex()));
+    }
+    else
```

Comment:
This else block is when the Exchange does not have a session at all, right? Would it still need to Encode? Should this be an outright failure?

Reply:
I have no idea 😬🤦‍♂️

It seemed like the infra knowing what fabric we're encoding data for is something we should always have available; but without this check, some tests were crashing

I don't understand why CommandHandlerImpl doesn't always have the fabric the command is scoped to.. filed https://github.com/project-chip/connectedhomeip/issues/39001